test/built-ins/Date/prototype/toString/non-date-receiver.js

```diff
@@ -0,0 +1,12 @@
+// Copyright (C) 2017 V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: #sec-date.prototype.tostring
+description: Date.prototype.toString throws a TypeError on non-Date receivers
+---*/
+
+assert.throws(TypeError, () => Date.prototype.toString());
+assert.throws(TypeError, () => Date.prototype.toString.call(undefined));
+assert.throws(TypeError, () => Date.prototype.toString.call(0));
+assert.throws(TypeError, () => Date.prototype.toString.call({}));
```

Comment:
I would like to suggest 2 special cases we can add in here:

```js
Date.prototype.toString.call("Tue Mar 21 2017 12:16:43 GMT-0400 (EDT)")
Date.prototype.toString.call(1490113011493)
```

It seems silly but they're still parseable values.